src/test/java/spoon/test/invocations/InvocationTest.java

```diff
@@ -58,4 +62,32 @@ public void testTargetNullForStaticMethod() throws Exception {
 			fail();
 		}
 	}
+
+	@Test
+	public void testInvocationWithArgument() throws Exception {
+		Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/java/spoon/test/invocations/testclasses/Foo.java");
+		launcher.buildModel();
+
+		CtClass<?> foo = launcher.getFactory().Class().get(Foo.class);
+		CtConstructor<?> constructor = foo.getConstructors().iterator().next();
+
+		CtInvocation<?> invocBar = constructor.getBody().getLastStatement();
+
+		assertEquals(3, invocBar.getArguments().size());
+		CtExpression secondArg = invocBar.getArguments().get(1);
+
+		assertTrue(secondArg instanceof CtLiteral);
+
+		CtLiteral arg = (CtLiteral)secondArg;
+		assertEquals(42, arg.getValue());
+
+		secondArg = launcher.getFactory().Core().createLiteral().setValue(12);
+
+		invocBar.addArgument(1, secondArg);
```

Comment:
> Interesting, this should fail with an exception because it's not compatible with the method signature.

If you do that, we should do it in all methods, especially if we create `addParameter(int position, ...)` in CtExecutable: then you'll have an egg-chicken problem, if you use a transformation on the executable and on the invocations, there will be a "not compatible transformation" somewhere.

So I agree with Thomas: we should not do check there, but we should document the fact that those transformation are not checked against the executable.